presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExpressionRewriteRuleSet.java

```diff
@@ -56,19 +58,28 @@
     }
 
     private final ExpressionRewriteRule rewriter;
+    private final Set<Rule<?>> rules;
 
-    private final ImmutableSet<Rule<?>> rules = ImmutableSet.of(
-            new ProjectExpressionRewrite(),
-            new AggregationExpressionRewrite(),
-            new FilterExpressionRewrite(),
-            new TableScanExpressionRewrite(),
-            new JoinExpressionRewrite(),
-            new ValuesExpressionRewrite(),
-            new ApplyExpressionRewrite());
+    private final ImmutableMap<Class<? extends PlanNode>, Rule<?>> allRules = ImmutableMap.<Class<? extends PlanNode>, Rule<?>>builder()
+            .put(ProjectNode.class, new ProjectExpressionRewrite())
+            .put(AggregationNode.class, new AggregationExpressionRewrite())
+            .put(FilterNode.class, new FilterExpressionRewrite())
+            .put(TableScanNode.class, new TableScanExpressionRewrite())
+            .put(JoinNode.class, new JoinExpressionRewrite())
+            .put(ValuesNode.class, new ValuesExpressionRewrite())
+            .put(ApplyNode.class, new ApplyExpressionRewrite())
```

Comment:
maybe make the rewrite classes public and static, and make a constructor of `ERRS` to accept a set of rules